lgc/test/Transforms/Continufy/simple.ll

```diff
@@ -0,0 +1,92 @@
+  %pushconst = call ptr addrspace(4) @lgc.user.data(i32 0)
+  %fn = load ptr, ptr addrspace(4) %pushconst
+  %p8 = getelementptr i8, ptr addrspace(4) %pushconst, i32 8
+  %x = load i32, ptr addrspace(4) %p8
+  %p16 = getelementptr i8, ptr addrspace(4) %pushconst, i32 16
+  %dst = load ptr addrspace(1), ptr addrspace(4) %p16
+  %r = call spir_func [2 x i32] %fn(i32 %x, ptr addrspace(1) %dst), !continufy.stage !{i32 -1}
+  store [2 x i32] %r, ptr addrspace(1) %dst
+  ret void
+}
+
+define spir_func i32 @chs(i32 %x) !lgc.shaderstage !{i32 7} !continufy.stage !{i32 3} {
+  %pushconst = call ptr addrspace(4) @lgc.user.data(i32 24)
+  %fn = load ptr, ptr addrspace(4) %pushconst
+  %y = call spir_func i32 %fn(i32 %x), !continufy.stage !{i32 5}
+  ret i32 %y
+}
+
+; Note: No !continufy.stage metadata here
+define dllexport void @lgc.shader.CS.main() !lgc.shaderstage !{i32 7} {
+entry:
+  %id = call i32 @lgc.shader.input.LocalInvocationId(i32 49)
+  %live = icmp ult i32 %id, 29
+  br i1 %live, label %main, label %exit
+
+main:
+  %pushconst = call ptr addrspace(4) @lgc.user.data(i32 32)
+  %fn = load ptr, ptr addrspace(4) %pushconst
+  call spir_func void %fn(), !continufy.stage !{i32 0}
+  br label %exit
+
+exit:
+  ret void
+}
+
+declare ptr addrspace(4) @lgc.user.data(i32)
+declare i32 @lgc.shader.input.LocalInvocationId(i32)
+; CHECK-LABEL: define {{[^@]+}}@raygen
+; CHECK-SAME: ({} [[STATE:%.*]], i32 [[RCR:%.*]]) !lgc.shaderstage !2 !continufy.stage !3 !lgc.cps !3 {
+; CHECK-NEXT:    [[PUSHCONST:%.*]] = call ptr addrspace(4) @lgc.user.data(i32 0)
+; CHECK-NEXT:    [[FN:%.*]] = load ptr, ptr addrspace(4) [[PUSHCONST]], align 8
+; CHECK-NEXT:    [[P8:%.*]] = getelementptr i8, ptr addrspace(4) [[PUSHCONST]], i32 8
+; CHECK-NEXT:    [[X:%.*]] = load i32, ptr addrspace(4) [[P8]], align 4
+; CHECK-NEXT:    [[P16:%.*]] = getelementptr i8, ptr addrspace(4) [[PUSHCONST]], i32 16
+; CHECK-NEXT:    [[DST:%.*]] = load ptr addrspace(1), ptr addrspace(4) [[P16]], align 8
+; CHECK-NEXT:    [[TMP1:%.*]] = ptrtoint ptr [[FN]] to i32
+; CHECK-NEXT:    [[TMP2:%.*]] = or i32 [[TMP1]], 2
+; CHECK-NEXT:    [[TMP3:%.*]] = call [2 x i32] (...) @lgc.cps.await.a2i32(i32 [[TMP2]], i32 4, i32 [[X]], ptr addrspace(1) [[DST]])
+; CHECK-NEXT:    store [2 x i32] [[TMP3]], ptr addrspace(1) [[DST]], align 4
+; CHECK-NEXT:    ret void
+;
+;
+; CHECK-LABEL: define {{[^@]+}}@chs
+; CHECK-SAME: ({} [[STATE:%.*]], i32 [[RCR:%.*]], i32 [[X:%.*]]) !lgc.shaderstage !2 !continufy.stage !4 !lgc.cps !5 {
+; CHECK-NEXT:    [[PUSHCONST:%.*]] = call ptr addrspace(4) @lgc.user.data(i32 24)
+; CHECK-NEXT:    [[FN:%.*]] = load ptr, ptr addrspace(4) [[PUSHCONST]], align 8
+; CHECK-NEXT:    [[TMP1:%.*]] = ptrtoint ptr [[FN]] to i32
+; CHECK-NEXT:    [[TMP2:%.*]] = or i32 [[TMP1]], 1
+; CHECK-NEXT:    [[TMP3:%.*]] = call i32 (...) @lgc.cps.await.i32(i32 [[TMP2]], i32 2, i32 [[X]])
+; CHECK-NEXT:    call void (...) @lgc.cps.jump(i32 [[RCR]], i32 5, {} poison, i32 poison, i32 [[TMP3]])
```

Comment:
Why is `levels = 5` here? Shouldn't it be 4 according to getReturnedLevels?

Reply:
I made some mistake to use `.ll` postfix as file name, and we don't include such test in "lgc/test/lit.cfg.py", I did not notice that the test was not run through `ninja check-lgc`. I am not sure whether it sounds reasonable to also include `.ll` test since we are using LLVM IR file for the test?